mesh/amqp.c

```diff
@@ -437,11 +446,53 @@ static void connect_with_delay(struct amqp_thread_context *context, int delay)
 		return;
 	}
 
-	l_info("%p", context);
-
 	set_amqp_state(context, MESH_AMQP_STATE_CONNECTING);
 }
 
+static bool amqp_subscribe_topic(const char *topic,
+					struct amqp_thread_context *context)
+{
+	amqp_rpc_reply_t reply;
+
+	char *queue_name = l_strdup_printf("rc.%s.raw", context->config.routing_key);
+
+	amqp_queue_bind(context->conn_state, 1,
+			amqp_cstring_bytes(queue_name),
+			amqp_cstring_bytes(context->config.exchange),
+			amqp_cstring_bytes(topic),
+			amqp_empty_table);
+
+	reply = amqp_get_rpc_reply(context->conn_state);
+	if (!is_reply_ok(&reply)) {
+		l_info("Failed to subscribe exchange topic: '%s'", topic);
+		l_free(queue_name);
+		return false;
+	}
+
+	l_info("Exchange topic: '%s' bound with queue: '%s'",
+							topic, queue_name);
+
+	l_free(queue_name);
+
+	return true;
+}
+
+
+static bool amqp_subscribe_topics(struct amqp_thread_context *context)
+{
+	const struct l_queue_entry *subscription =
+				l_queue_get_entries(context->subscriptions);
+
+	while (subscription) {
```

Comment:
`l_queue_foreach`

Reply:
`l_queue_foreach` doesn't allow to return value. So I can't break the loop when one of subscriptions will fail.